java/src/main/kotlin/io/spine/validation/java/setonce/SetOnceStringField.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2024, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validation.java.setonce
+
+import com.intellij.psi.PsiBlockStatement
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiIfStatement
+import io.spine.protodata.ast.Field
+import io.spine.protodata.ast.PrimitiveType
+import io.spine.protodata.java.AnElement
+import io.spine.protodata.java.Expression
+import io.spine.protodata.type.TypeSystem
+import io.spine.tools.psi.java.method
+
+/**
+ * Renders Java code to support `(set_once)` option for the given string [field].
+ *
+ * @param field The string field that declared the option.
+ * @param typeSystem The type system to resolve types.
+ */
+internal class SetOnceStringField(
+    field: Field,
+    typeSystem: TypeSystem
+) : SetOnceJavaConstraints<String>(field, typeSystem) {
+
+    init {
+        check(field.type.primitive == PrimitiveType.TYPE_STRING) {
+            "`${javaClass.simpleName}` handles only number fields. " +
```

Comment:
We compare the field type with `string`, and if it is not one, we say that this field handles only **number** fields. Is that really so?